aurora/src/main/java/com/market/banica/aurora/config/Publishers.java

```diff
@@ -0,0 +1,106 @@
+package com.market.banica.aurora.config;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.market.banica.common.util.ApplicationDirectoryUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.EnableMBeanExport;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedResource;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+@EnableMBeanExport
+@ManagedResource
+@Configuration
+public class Publishers {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Publishers.class);
+
+    private final String publishersFileName;
+
+    CopyOnWriteArrayList<String> publishersList;
+
+
+    @Autowired
+    public Publishers(@Value("${aurora.channels.publishers}") String fileName) {
+        this.publishersFileName = fileName;
+        this.publishersList = this.readPublishersFromFile();
+    }
+
+    @ManagedOperation
+    public List<String> getPublishersList() {
+        return publishersList;
+    }
+
+    @ManagedOperation
+    public void addPublisher(String publisher) {
+        if (publishersList.contains(publisher)) {
+            throw new IllegalArgumentException("Publisher is already defined.");
+        }
+        publishersList.add(publisher);
+        writeBackUp();
+    }
+
+    @ManagedOperation
+    public void deletePublisher(String publisher) {
+        if (!publishersList.contains(publisher)) {
+            throw new IllegalArgumentException("Publisher is not defined.");
+        }
+        publishersList.remove(publisher);
+        writeBackUp();
+    }
+
+    protected void writeBackUp() {
+        LOGGER.info("Writing back-up to json");
```

Comment:
add some synchronization here, this file can be read/write asynchronously if we have 2 users using jmx

Reply:
Fixed.